infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/util/ReflectionHelper.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.ogm.datastore.infinispan.util;
+
+import java.beans.BeanInfo;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Reflection utilities.
+ *
+ * @author The Viet Nguyen &amp;ntviet18@gmail.com&amp;
+ */
+public class ReflectionHelper {
+
+	/**
+	 * Introspect the given object.
+	 *
+	 * @param obj object for introspection.
+	 *
+	 * @return a map containing object's field values.
+	 *
+	 * @throws IntrospectionException if an exception occurs during introspection
+	 * @throws InvocationTargetException if property getter throws an exception
+	 * @throws IllegalAccessException if property getter is inaccessible
+	 */
+	public static Map<String, Object> introspect(Object obj) throws IntrospectionException, InvocationTargetException, IllegalAccessException {
+		Map<String, Object> result = new HashMap<>();
+		BeanInfo info = Introspector.getBeanInfo( obj.getClass() );
+		for ( PropertyDescriptor pd : info.getPropertyDescriptors() ) {
+			Method reader = pd.getReadMethod();
+			String name = pd.getName();
+			if ( reader != null && !"class".equals( name ) ) {
+				result.put( name, reader.invoke( obj ) );
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * Load and instantiate given type.
+	 *
+	 * @param className full qualified class name
+	 *
+	 * @return new object of given class.
+	 *
+	 * @throws ClassNotFoundException if the class cannot be located
+	 * @throws IllegalAccessException if class or its no-arg constructor are not accessible
+	 * @throws InstantiationException if the class cannot be instantiated, e.g. abstract class, interface, etc.
+	 */
+	public static Object instantiate(String className) throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+		Class<?> clazz = Class.forName( className );
```

Comment:
This would prevent OGM to work correctly in modular environments. When needing to create a new instance of a user domain model by name, we should use `ClassLoaderService`.

You can find several examples of using the service in the code base. For example `org.hibernate.ogm.service.impl.SessionFactoryEntityNamesResolver` does something similar.